extensions/amp-ad/0.1/amp-ad-api-handler.js

```diff
@@ -143,25 +156,51 @@ export class AmpAdApiHandler {
 
   /**
    * Updates the element's dimensions to accommodate the iframe's
-   *    requested dimensions.
+   * requested dimensions. Notifies the window that request the resize
+   * of success or failure.
    * @param {number|undefined} height
    * @param {number|undefined} width
+   * @param {!Window} source
+   * @param {string} origin
    * @private
-   */
-  updateSize_(height, width) {
+  */
+  updateSize_(height, width, source, origin) {
+    if (this.pendingResizeRequest_) {
+      // There is an already pending resize request, fail it.
+      this.sendEmbedSizeResponse_(false /* success */);
+    }
```

Comment:
This refactor is breaking the existing resize functionality. 
I am going to revert this for now before we cut the next canary. 

cc @lannka @zhouyx 

Please reach out to @lannka/myself for more details